terraform/20-app/elasticache.front-end.tf

```
@@ -0,0 +1,25 @@
+module "front_end_elasticache_security_group" {
+  source  = "terraform-aws-modules/security-group/aws"
+  version = "5.1.0"
+
+  name   = "${local.prefix}-front-end"
+  vpc_id = module.vpc.vpc_id
+
+  ingress_with_source_security_group_id = [
+    {
+      description              = "frontend tasks to cache"
+      rule                     = "redis-tcp"
+      source_security_group_id = module.ecs_service_front_end.security_group_id
+    }
+  ]
+}
+
+resource "aws_elasticache_serverless_cache" "front_end_elasticache" {
+  engine = "redis"
+  name   = "${local.prefix}-front-end"
+
+  major_engine_version     = "7"
+  snapshot_retention_limit = 1
+  security_group_ids       = [module.front_end_elasticache_security_group.security_group_id]
+  subnet_ids               = module.vpc.elasticache_subnets
+}
```

Do we want to add more here to enable authentication for Redis and encrypt data in transit e.g.
```suggestion
  transit_encryption_enabled   = true
  auth_token                   = var.redis_auth_token

  log_delivery_configuration {
    destination_type = "cloudwatch-logs"
    log_format       = "json"
    log_type         = "engine-log"
  }
}
```

---

@ChristianAMartin this feels like scope creep, I'd say we can pick this up in a follow up PR

---

yes agreed. Happy to do this.